ext/liquid_c/vm.c

```diff
@@ -232,6 +233,7 @@ static VALUE vm_render_until_error(VALUE uncast_args)
 {
     vm_render_until_error_args_t *args = (void *)uncast_args;
     const VALUE *const_ptr = args->const_ptr;
+    const VALUE *tags_ptr = args->tags_ptr;
```

Comment:
I think you also need add a corresponding `args->tags_ptr = tags_ptr;` at the end of `case OP_RENDER_VARIABLE_RESCUE`.